core/src/main/resources/hudson/model/AbstractBuild/summary.jelly

```diff
@@ -0,0 +1,72 @@
+<!--
+The MIT License
+
+Copyright (c) 2004-2009, Sun Microsystems, Inc., Kohsuke Kawaguchi, Matthew R. Harrah,
+Tom Huybrechts, id:cactusman, Romain Seguy
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:t="/lib/hudson" xmlns:l="/lib/layout">
+  <j:set var="set" value="${it.changeSet}" />
+  <t:summary icon="symbol-changes">
+    <j:choose>
+      <j:when test="${it.hasChangeSetComputed()}">
+        <st:include it="${set}" page="digest.jelly" />
```

Comment:
Let me know if I'm misunderstanding -  Add a file that uses the updated changesets method, then `<st:include ... />` it in both `AbstractBuild/main.jelly` and `WorkflowRun/main.jelly`?

Reply:
I've added `<st:include page="changesets.jelly" class="jenkins.scm.RunWithSCM" />` to `summary.jelly` in `AbstractBuild` and `WorkflowRun`. The implementation of changesets is different in the two files (as you noted above) with one difference being that `WorkflowRun` lacks 

```
<!-- dependency changes -->
              <j:set var="depChanges" value="${it.getDependencyChanges(it.previousBuild)}"/>
              <j:if test="${!empty(depChanges)}">
                ${%Changes in dependency}
                <ol>
                  <j:forEach var="dep" items="${depChanges.values()}">
                    <li>
                      <a href="${rootURL}/${dep.project.url}" class="model-link">${dep.project.displayName}</a>
                      <st:nbsp/>
                      <j:choose>
                        <j:when test="${dep.from!=null}">
                          <a href="${rootURL}/${dep.from.url}" class="model-link inside">
                            <l:icon class="${dep.from.buildStatusIconClassName} icon-sm" alt="${dep.from.iconColor.description}"/>${dep.from.displayName}</a>
                        </j:when>
                        <j:otherwise>
                          ?
                        </j:otherwise>
                      </j:choose>

                      &#x2192; <!-- right arrow -->

                      <a href="${rootURL}/${dep.to.url}" class="model-link inside">
                        <l:icon class="${dep.to.buildStatusIconClassName} icon-sm" alt="${dep.to.iconColor.description}" />${dep.to.displayName}</a>

                      (<a href="${rootURL}/${dep.project.url}changes?from=${dep.fromId}&amp;to=${dep.toId}">${%detail}</a>)
                    </li>
                  </j:forEach>
                </ol>
              </j:if>
```

\-   Would this be needed in `changesets.jelly`?

Reply:
I am not very familiar with this but looks like something never implemented in Pipeline.

In general, for a view refactoring like this it is necessary to examine every line in all variants, figure out what it is doing and where it came from (was this a copy of an old version of something that was since updated at the source?), and decide which parts can be made common without loss of functionality and which intrinsically need customization.